server/server.go

```diff
@@ -642,8 +643,8 @@ func (s *Server) parsePublishParams(r *http.Request, v *visitor, m *message) (ca
 			return false, false, "", false, errHTTPBadRequestDelayTooLarge
 		}
 		m.Time = delay.Unix()
-		m.Sender = v.ip // Important for rate limiting
 	}
+	m.Sender = v.ip // Important for rate limiting
```

Comment:
This was because of https://github.com/binwiederhier/ntfy/pull/429#discussion_r988526622. I thought there was a different bug, I'll make that not must and make this not record the IP for every message.